readme.md

```diff
@@ -3,71 +3,704 @@
 
-[NPM](https://www.npmjs.com/package/grunt)
+[Solution Code](https://github.com/ga-wdi-exercises/gulp-bamsay/tree/jshint-solution)
+
+### Installation
 
 ```bash
-$ npm install -g grunt-cli
+$ npm install gulp -g
 ```
+
+>install globally
+
 ```bash
-$ npm install grunt --save
+$ npm install gulp --save-dev
 ```
+>install as dependency in project
+
 
 ```bash
-$ touch Gruntfile.js
+$ touch `gulpfile.js`
 ```
 >This needs to be installed in your root directory and will contain all of your related task configuration
 
+```bash
+$ gulp
+
+```
+```bash
+
+[10:36:31] Using gulpfile ~/wdi8/lessons/build-tools/gulp-example/gulpfile.js
+[10:36:32] Task 'default' is not in your gulpfile
+[10:36:32] Please check the documentation for proper gulpfile formatting
+
+
+```
+>Uh oh! - we ran into a problem! We need to define a `gulp task`
+
+### Gulp Tasks
+
+<details>
+<summary>
+Q. What is a task?</summary>
+<br>
+
+```
+
+In Gulp, we create tasks that can transform our code.
+A task may perform one job, or it can perform many at once.
+
+```
+<br>
+<br>
+</details>
 
-[Grunt plugins](http://gruntjs.com/plugins)
+----------
 
+In our `gulpfile.js` we need to include the gulp module. To do this, we should define a variable: `var gulp = require('gulp');` This will allow us to call upon Gulp to create a task.
 
->Each time grunt is run, it looks for a locally installed Grunt using node's require() system. Because of this, you can run grunt from any subfolder in your project.
+### My First (Default) Task
 
+By default, Gulp requires a default task. It is the first task that Gulp will look for when reading your `gulpfile.js`.
 
-In our Gruntfile:
+Let's define our first (default) task below in our `gulpfile.js`:
 
 ```js
-module.exports = function(grunt) {
+var gulp = require('gulp');
 
-  grunt.initConfig({
+//define a task with the name of 'default'
+// and a callback to perform when the task is ran
+gulp.task('default', function() {
+  console.log('I am the default task!');
+});
 
-    pkg: grunt.file.readJSON('package.json'),
-      jshint: {
-        all:  ['Gruntfile.js', 'app/**/*.js', '*.js']
-      }
+```
+In your terminal, run `$ gulp`.
 
-  });
+>This will have the library look for a default task in your `gulpfile.js`. It will then execute the callback that you define for your task.
+
+You should see something like the following:
+
+```bash
+[10:37:37] Using gulpfile ~/wdi8/lessons/build-tools/gulp-example/gulpfile.js
+[10:37:37] Starting 'default'...
+I am the default task!
+[10:37:37] Finished 'default' after 104 μs
+```
+
+<details>
+<summary>
+Q. Does the string `'default'` matter what we call it?
+</summary>
+<br>
+
+```
+Yes, and no. While you can certainly change `default` to `pizza`, it would not be very descriptive of the task.
+
+Also, whatever you define 'name' as in your task: gulp.task('name', callbackFunction)
+
+You need to run the following in our command line:
+
+$ gulp <name>
```

Comment:
could be an opportunity for a you do, e.g. create a new task named `today` or something and print today's date